src/renderers/webgl/edge.ts

```diff
@@ -1,46 +1,52 @@
-import { DEFAULT_LABEL_STYLE, MIN_EDGES_ZOOM, MIN_INTERACTION_ZOOM, MIN_LABEL_ZOOM } from '../../utils/constants'
-import { type Renderer } from '.'
-import { midPoint } from './utils'
-import { movePoint } from './utils'
-import { NodeRenderer } from './node'
-import type { ArrowStyle, Edge } from '../../types'
-import { Arrow } from './objects/arrow'
-import { LineSegment } from './objects/lineSegment'
+import {
+  DEFAULT_ARROW,
+  DEFAULT_LABEL_STYLE,
+  MIN_EDGES_ZOOM,
+  MIN_INTERACTION_ZOOM,
+  MIN_LABEL_ZOOM,
+  MIN_STROKE_ZOOM
+} from '../../utils/constants'
+import type { ArrowStyle, Edge, PointTuple } from '../../types'
 import { FederatedPointerEvent } from 'pixi.js'
+import { midPoint, movePoint } from '../../utils/webgl'
+import { angle, distance } from '../../utils/api'
+import { NodeRenderer } from './node'
+import { type Renderer } from '.'
 import { EdgeHitArea } from './interaction/edgeHitArea'
-import { angle } from '../../utils/api'
+import Arrow from './objects/Arrow'
+import LineSegment from './objects/line/LineSegment'
+import LineStrokes from './objects/line/LineStrokes'
 import Text from './objects/text/Text'
 
-const DEFAULT_EDGE_WIDTH = 1
-const DEFAULT_EDGE_COLOR = 0xaaaaaa
-const DEFAULT_ARROW = 'none'
-
 export class EdgeRenderer {
   edge!: Edge
   source!: NodeRenderer
   target!: NodeRenderer
-  label?: Text
-  renderer: Renderer
-  lineSegment: LineSegment
-  x0 = 0
-  y0 = 0
-  x1 = 0
-  y1 = 0
-  theta = 0
-  center: [x: number, y: number] = [0, 0]
-  width?: number
-  stroke?: string | number
-  strokeOpacity?: number
   sourceRadius?: number
   targetRadius?: number
 
+  private x0 = 0
+  private y0 = 0
+  private x1 = 0
+  private y1 = 0
+  private length = 0
+  private theta = 0
+  private center: PointTuple = [0, 0]
+  private lineSegment: LineSegment
   private hitArea: EdgeHitArea
+  private strokes?: LineStrokes
```

Comment:
Is the new LineStrokes object used to support edges with multiple strokes, so an extension of the edge style spec (which currently only supports a single line)? Cool if so.

Reply:
yes exactly. There is probably some fine tuning to be done here, such as wrapping additional strokes around arrows etc.